dataloader/dataloader.hpp

```diff
@@ -250,6 +380,87 @@ class DataLoader
     return index;
   }
 
+  /**
+   * Performs train test split.
+   *
+   * @tparam DatasetType Type of input dataset.
+   * @tparam LabelsType Type of input labels.
+   *
+   * @param dataset Features of dataset.
+   * @param labels Labels of the dataset.
+   * @param validRatio Ratio for train-test split.
+   * @param shuffle Boolean to determine shuffling of dataset.
+   */
+  void TrainTestSplit(DatasetX dataset,
+                      std::deque<arma::vec> labels,
+                      DatasetX& trainFeatures,
+                      arma::field<arma::vec>& trainLabels,
+                      DatasetX& validFeatures,
+                      arma::field<arma::vec>& validLabels,
+                      const double validRatio,
+                      const bool shuffle)
+  {
+    const size_t validSize = static_cast<size_t>(dataset.n_cols * validRatio);
+    const size_t trainSize = dataset.n_cols - validSize;
+
+    std::cout << trainSize << " " << validSize << std::endl;
+    arma::uvec order = arma::linspace<arma::uvec>(0, dataset.n_cols - 1,
+      dataset.n_cols);
+    if (shuffle)
+    {
+      order = arma::shuffle(order);
+    }
+
+    if (trainSize > 0)
+    {
+      trainLabels.set_size(1, trainSize);
+      trainFeatures = dataset.cols(order.subvec(0, trainSize - 1));
+
+      // Field type has fixed size so we can't use span and assignment
+      // operator.
+      for (size_t i = 0; i < trainSize; i++)
+      {
+        trainLabels(0, i) = std::move(labels[i]);
+      }
+    }
+
+    if (validSize < dataset.n_cols)
```

Comment:
When `validRatio` is 1, the `validSize` should be the same as `dataset.n_cols`
```suggestion
    if (validSize <= dataset.n_cols)
```

Reply:
Makes sense. I'll add some boundary checks as well.

Reply:
It seems you missed this comment, can you check it?

Reply:
Terribly sorry about that. Let me fix it in the next commit.

Reply:
Done. Added a boundary check as well [here](https://github.com/mlpack/models/pull/18/files#diff-e334979f69acae3640bee0da79a64c3aR162). Sorry I missed it before. Thanks.